### ISSUE TITLE
Smooth layout changes with Transition API

### ISSUE THREAD

**Issue by todorone** [software-mansion/react-native-reanimated#264]
New `Transition` API looks like great alternative to `LayoutAnimation` API, but when elements appear/disappear, we do not have smooth changes of other elements, that being affected via layout changes, even with `scale` type. In case of `LayoutAnimation` there is no such a problems and also in case of `Transition.Change`, we also have smooth experience, the only problem is elements' appearance/disappearance.

Any ideas on how we can mimic `LayoutAnimation`'s behavior with new `Transition` API?

**Comment by WoLewicki** [comment]
Is it still the case? If yes, could you provide a code example?

**Comment by todorone** [comment]
Thanks for attention. Sure, will try to find time on a weekend to create repro snack.

**Comment by todorone** [comment]
Interesting, tried to reproduce an issue, got another one. 😛 
That's the Snack - if tapped on `Animate` button twice, the title of it is disappeared and it becomes unresponsive for taps. GIF attached.

https://snack.expo.io/@todorone/1647ca

![reanimated-issue](https://user-images.githubusercontent.com/423701/71320322-5fe76400-24b2-11ea-9219-3ccf68230a14.gif)

**Comment by calvinaquino** [comment]
I have the same issue.

I have the app content and below it, a bar that moves into the screen, occupying space.
This makes the app content resize it's content to accommodate the bar on the screen.

While the bar animation with `Transition` is perfect, the content just instantly snaps into it's new size.

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.
 
We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
